core/src/main/java/com/valtech/aem/saas/core/autocomplete/AutocompleteServlet.java

```diff
@@ -43,31 +42,36 @@
   @Override
   protected void doGet(@NonNull SlingHttpServletRequest request,
       @NonNull SlingHttpServletResponse response) throws ServletException, IOException {
-
-    Optional.ofNullable(request.adaptTo(RequestWrapper.class))
-        .ifPresent(requestWrapper ->
-            requestWrapper.getParameter(SearchTabModelImpl.SEARCH_TERM).ifPresent(text -> {
-              SearchConfiguration searchConfiguration = request.getResource().adaptTo(ConfigurationBuilder.class)
-                  .as(SearchConfiguration.class);
-              if (StringUtils.isNotBlank(searchConfiguration.index())) {
-                TypeaheadPayloadDTO payload = DefaultTypeaheadPayloadDTO.builder()
-                    .text(text)
-                    .language(getLanguage(request).orElse(StringUtils.EMPTY))
-                    .filters(getSearch(request).map(SearchModel::getEffectiveFilters).orElse(Collections.emptySet()))
-                    .build();
-                List<String> results = typeaheadService.getResults(searchConfiguration.index(), payload);
-                new JsonResponseCommitter(response).flush(printWriter -> new Gson().toJson(results, printWriter));
-              }
-            }));
+    RequestWrapper requestWrapper = request.adaptTo(RequestWrapper.class);
+    if (requestWrapper == null) {
+      throw new IllegalArgumentException("Can not adapt the request to RequestWrapper sling model.");
+    }
+    String searchTerm = requestWrapper.getParameter(SearchTabModelImpl.SEARCH_TERM)
+        .orElseThrow(() -> new IllegalArgumentException("Search term not specified."));
+    SearchCAConfigurationModel searchCAConfigurationModel = Optional.ofNullable(request.getResource()
+            .adaptTo(SearchCAConfigurationModel.class))
+        .orElseThrow(
+            () -> new IllegalArgumentException("Could not access search CA configurations from current resource."));
+    List<String> results = typeaheadService.getResults(searchCAConfigurationModel, searchTerm,
+        getLanguage(request),
+        getPredefinedSearchFilters(request));
+    new JsonResponseCommitter(response).flush(printWriter -> new Gson().toJson(results, printWriter));
   }
 
+  private Set<FilterModel> getPredefinedSearchFilters(@NonNull SlingHttpServletRequest request) {
+    return getSearch(request)
+        .map(SearchModel::getEffectiveFilters)
+        .orElse(Collections.emptySet());
+  }
 
   private Optional<SearchModel> getSearch(@NonNull SlingHttpServletRequest request) {
     return Optional.ofNullable(request.getResource().adaptTo(SearchModel.class));
   }
 
-  private Optional<String> getLanguage(@NonNull SlingHttpServletRequest request) {
-    return Optional.ofNullable(request.getResource().adaptTo(ResourceWrapper.class)).map(ResourceWrapper::getLocale)
-        .map(Locale::getLanguage);
+  private String getLanguage(@NonNull SlingHttpServletRequest request) {
```

Comment:
we should get the Language from the SearchModel if possible. Reason: We will need to make the language a configurable paramater. If empty it will fall back to what you have now

Reply:
langauge is taken from request's resource.
in this case, the resource is the search component.